services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -114,17 +114,17 @@ class HankeController(
     fun deleteHanke(@PathVariable hankeTunnus: String) {
         logger.info { "Deleting hanke: $hankeTunnus" }
 
-        val hanke = hankeService.loadHanke(hankeTunnus) ?: throw HankeNotFoundException(hankeTunnus)
-        val hankeId = hanke.id!!
-
-        val userId = currentUserId()
-        if (!permissionService.hasPermission(hankeId, userId, PermissionCode.DELETE)) {
-            throw HankeNotFoundException(hankeTunnus)
+        hankeService.getHankeHakemuksetPair(hankeTunnus).let {
+            val hanke = it.first
+            val hakemukset = it.second
```

Comment:
I find it easier to read if pairs are destructured on use:
```suggestion
        hankeService.getHankeHakemuksetPair(hankeTunnus).let { (hanke, hakemukset) =>
```

Reply:
Fair point, changed as suggested.